test/ai-jsx.spec.tsx

```diff
@@ -0,0 +1,210 @@
+import axios from 'axios';
+import * as AI from 'ai-jsx';
+import {
+  ChatCompletion,
+  UserMessage,
+  SystemMessage,
+} from 'ai-jsx/core/completion';
+import {
+  AutoblocksJsxTracer,
+  AutoblocksPlaceholder,
+} from '../src/ai-jsx/index';
+
+interface SentEvent {
+  message: string;
+  traceId: string;
+  timestamp: string;
+  properties: {
+    spanId: string;
+    parentSpanId?: string;
+    latency?: number;
+    [key: string]: unknown;
+  };
+}
+
+jest.setTimeout(100000);
+jest.mock('axios');
+
+const axiosCreateMock = axios.create as jest.Mock;
+
+const expectAllEqualAndDefined = (xs: string[]) => {
+  expect(xs.every((x) => x === xs[0])).toBe(true);
+  expect(xs.every(Boolean)).toBe(true);
+};
+
+describe('ai-jsx', () => {
+  beforeAll(() => {
+    process.env.AUTOBLOCKS_INGESTION_KEY = 'test';
+  });
+
+  afterAll(() => {
+    process.env.AUTOBLOCKS_INGESTION_KEY = undefined;
+  });
+
+  let mockPost: jest.Mock;
+
+  beforeEach(() => {
+    mockPost = jest
+      .fn()
+      .mockResolvedValue({ data: { traceId: 'mock-trace-id' } });
+    axiosCreateMock.mockReturnValueOnce({ post: mockPost });
+  });
+
+  const sentEvents = () => mockPost.mock.calls.map((c) => c[1]) as SentEvent[];
+
+  const makeSpanPairs = () => {
+    const events = sentEvents();
+    const pairs: Record<string, SentEvent[]> = {};
+    for (const event of events) {
+      if (!pairs[event.properties.spanId]) {
+        pairs[event.properties.spanId] = [];
+      }
+      pairs[event.properties.spanId].push(event);
+    }
+    return pairs;
+  };
+
+  afterEach(() => {
```

Comment:
it feels a little awkward to have all these expects in the afterEach, I think it'd be cleaner to have an expect function we call in the its

Reply:
but then you need to remember to call it in each test

Reply:
ok that makes sense, i'll pull into a function and call at the end

Reply:
i tried pulling everything out of afterEach and into a function but `mockPost` is undefined when calling the function for some reason

```
    TypeError: Cannot read properties of undefined (reading 'mock')

      52 |   });
      53 |
    > 54 |   const sentEvents = () => mockPost.mock.calls.map((c) => c[1]) as SentEvent[];
```